gstreamer-rtsp-server/src/auto/r_t_s_p_thread_pool.rs

```diff
@@ -0,0 +1,92 @@
+// This file was generated by gir (37f7805+) from gir-files (???)
+// DO NOT EDIT
+
+use ffi;
+use glib;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct RTSPThreadPool(Object<ffi::GstRTSPThreadPool, ffi::GstRTSPThreadPoolClass>);
+
+    match fn {
+        get_type => || ffi::gst_rtsp_thread_pool_get_type(),
+    }
+}
+
+impl RTSPThreadPool {
+    pub fn new() -> RTSPThreadPool {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_rtsp_thread_pool_new())
+        }
+    }
+
+    pub fn cleanup() {
+        assert_initialized_main_thread!();
+        unsafe {
+            ffi::gst_rtsp_thread_pool_cleanup();
+        }
+    }
+}
+
+impl Default for RTSPThreadPool {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+unsafe impl Send for RTSPThreadPool {}
+unsafe impl Sync for RTSPThreadPool {}
+
+pub trait RTSPThreadPoolExt {
+    fn get_max_threads(&self) -> i32;
+
+    //fn get_thread(&self, type_: RTSPThreadType, ctx: /*Ignored*/&mut RTSPContext) -> /*Ignored*/Option<RTSPThread>;
+
+    fn set_max_threads(&self, max_threads: i32);
+
+    fn connect_property_max_threads_notify<F: Fn(&Self) + Send + Sync + 'static>(&self, f: F) -> SignalHandlerId;
+}
+
+impl<O: IsA<RTSPThreadPool> + IsA<glib::object::Object>> RTSPThreadPoolExt for O {
+    fn get_max_threads(&self) -> i32 {
+        unsafe {
+            ffi::gst_rtsp_thread_pool_get_max_threads(self.to_glib_none().0)
+        }
+    }
+
+    //fn get_thread(&self, type_: RTSPThreadType, ctx: /*Ignored*/&mut RTSPContext) -> /*Ignored*/Option<RTSPThread> {
```

Comment:
We will definitely need the context at some point, but let's ignore that for now. I think we can get away with a manual implementation for the whole context and setting it to ref_mode ref-mut